src/org/traccar/protocol/LdplProtocolDecoder.java

```diff
@@ -0,0 +1,107 @@
+package org.traccar.protocol;
+
+import java.net.SocketAddress;
+import java.util.regex.Pattern;
+
+import org.jboss.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.Protocol;
+import org.traccar.helper.DateBuilder;
+import org.traccar.helper.Parser;
+import org.traccar.helper.Parser.CoordinateFormat;
+import org.traccar.helper.PatternBuilder;
+import org.traccar.model.Position;
+
+public class LdplProtocolDecoder extends BaseProtocolDecoder {
+
+    public LdplProtocolDecoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    private static final Pattern PATTERN = new PatternBuilder()
+                .text("*ID")                         // start of frame
```

Comment:
I think there is too much indentation. Please match other protocols. Also, please remove underscore in comments.